### ISSUE TITLE
Improve UX for creating a Blazor Web App when selecting None for the interactivity type

### ISSUE THREAD

**Issue by danroth27** [dotnet/templating#7092]
When creating a Blazor Web App, if you select None for the interactivity type then you can still select an interactivity location, which doesn't make any sense. Is there some way to disable the interactivity location option when the interactivity type is None? On the CLI should we make this an error?

**Comment by Eilon** [comment]
We have an additional similar need in an upcoming Blazor Hybrid + Web project template scenario.

I want to have a UI in VS like this:

Create new Blazor Hybrid project
* ✅ Include Blazor Web project with shared Razor UI
* ✅ Configure for HTTPS (but ONLY enabled if previous checkbox is checked!)
* ... and more options

In VS it would look like this when the top checkbox is checked:

![image](https://github.com/dotnet/templating/assets/202643/d6489d42-0654-44e7-9dc5-0c43de785035)

And when it's unchecked:

![image](https://github.com/dotnet/templating/assets/202643/bcff4e78-0e93-4513-b096-443b05172788)


As far as `template.json` syntax, I think it already exists, but apparently isn't supported in VS? See: https://github.com/dotnet/templating/issues/5648#issuecomment-1322400549, https://github.com/dotnet/templating/issues/3341

And it seems Blazor already uses `isEnabled` in its templates for this scenario: https://github.com/dotnet/aspnetcore/blob/d7109b13bcc87601227ab4d643638dbeb6a87295/src/ProjectTemplates/Web.ProjectTemplates/content/BlazorWeb-CSharp/.template.config/template.json

**Comment by baronfel** [comment]
cc @phenning for the VS template engine host request - conditional symbols would be the solution here but they are only supported in the CLI at the moment.

**Comment by Eilon** [comment]
Yup I just chatted with @phenning earlier today about this. We have workarounds for now but I think the best customer UI for our scenarios would require this new feature.

**Comment by Arclight3** [comment]
I also need this VS feature. I have a custom template and I want to hide or disable certain parameters based on a checkbox. It's a really bad user experience without this ability.

**Comment by Banner-Keith** [comment]
@baronfel is there a different team we need to complain to for the VS pieces?

**Comment by baronfel** [comment]
Submitting feedback via the [VS Developer Community](https://developercommunity.visualstudio.com/VisualStudio/suggest) would be the way to go. That'll get routed to @phenning and the rest of the team and can be used to help prioritize their backlog.

**Comment by Banner-Keith** [comment]
@baronfel I found this one. I am assuming this covers this scenario and we should just go vote for it>
https://developercommunity.visualstudio.com/t/conditional-parameter--hideshow-fields-/10254783

**Comment by danroth27** [comment]
@sayedihashimi @vijayrkn What's the best way for us to make progress on this issue? Are you already tracking https://developercommunity.visualstudio.com/t/conditional-parameter--hideshow-fields-/10254783 on the VS side?

**Comment by sayedihashimi** [comment]
@danroth27 we have move the item to our team. We will investigate. Hopefully this will be easy to implement, otherwise we may need to wait on adding this support. We will take a look to see how much effort this will take and then go from there.

Should we consider this an ask from the ASP.NET team to get this support added?

**Comment by danroth27** [comment]
> Should we consider this an ask from the ASP.NET team to get this support added?

Yes, this is needed to improve the user experience for both the Blazor Web App template and the upcoming Blazor Hybrid + Web template.

**Comment by sayedihashimi** [comment]
@danroth27 thanks, we are investigating.

**Comment by prince272** [comment]
> @danroth27 thanks, we are investigating.

How far with the investigation?
